core/services/vrf/v2/listener_v2.go

```diff
@@ -1557,28 +1587,226 @@ func (lsn *listenerV2) runRequestHandler(pollPeriod time.Duration, wg *sync.Wait
+		}
+	}
+	for _, l := range fulfillments {
+		if l.EventSig == lsn.coordinator.RandomWordsFulfilledTopic() {
+			parsed, err2 := lsn.coordinator.ParseRandomWordsFulfilled(l.ToGethLog())
+			if err2 != nil {
+				errs = multierr.Append(errs, err2)
+				continue
 			}
+			fulfilled[parsed.RequestID().String()] = parsed
+		} else {
+			errs = multierr.Append(errs, errors.Errorf("unexpected event sig %s", l.EventSig))
 		}
 	}
+	if err != nil {
+		return 0, errors.Wrap(err, "LogPoller.LogsCreatedAfter")
+	}
+	// figure out the earliest block number with an unfulfilled request
+	var unfulfilled []RandomWordsRequested
+	for reqID, req := range requested {
+		if _, ok := fulfilled[reqID]; !ok {
+			unfulfilled = append(unfulfilled, req)
+		}
+	}
+	var earliestUnfulfilledBlock int64
+	for _, req := range unfulfilled {
+		if earliestUnfulfilledBlock != 0 && req.Raw().BlockNumber < uint64(earliestUnfulfilledBlock) {
+			earliestUnfulfilledBlock = int64(req.Raw().BlockNumber)
+		} else {
+			// first iteration
+			earliestUnfulfilledBlock = int64(req.Raw().BlockNumber)
+		}
+	}
+	if earliestUnfulfilledBlock == 0 {
+		// no unfulfilled requests
+		return latestBlock.FinalizedBlockNumber, nil
+	}
+	return earliestUnfulfilledBlock, nil
+}
+
+func (lsn *listenerV2) updateLastProcessedBlock() (lastProcessedBlock int64, err error) {
+	lsn.reqsMu.Lock()
+	defer lsn.reqsMu.Unlock()
+	latestBlock, err := lsn.chain.LogPoller().LatestBlock()
+	if err != nil {
+		lsn.l.Errorw("error getting latest block", "err", err)
+		return 0, errors.Wrap(err, "LogPoller.LatestBlock()")
+	}
+	var earliestUnprocessedRequestBlock int64
+	for _, req := range lsn.reqs {
+		if earliestUnprocessedRequestBlock != 0 && req.req.Raw().BlockNumber < uint64(earliestUnprocessedRequestBlock) {
+			earliestUnprocessedRequestBlock = int64(req.req.Raw().BlockNumber)
+		} else {
+			// first iteration
+			earliestUnprocessedRequestBlock = int64(req.req.Raw().BlockNumber)
+		}
+	}
+	if earliestUnprocessedRequestBlock == 0 {
+		// no unprocessed requests
+		return latestBlock.FinalizedBlockNumber, nil
+	}
+	// whether the earliest unprocessed is before or after the finalized block doesn't
+	// matter.
+	return earliestUnprocessedRequestBlock, nil
+}
+
+// pollLogs uses the log poller to poll for the latest VRF logs
+func (lsn *listenerV2) pollLogs(ctx context.Context, minConfs uint32, lastProcessedBlock int64) error {
+	lp := lsn.chain.LogPoller()
+
+	// latest unfinalized block used on purpose to get bleeding edge logs
+	// we don't really have the luxury to wait for finalization on most chains
+	// if we want to fulfill on time.
+	latestBlock, err := lp.LatestBlock()
+	if err != nil {
+		return errors.Wrap(err, "LogPoller.LatestBlock()")
+	}
+	lsn.setLatestHead(latestBlock)
+
+	// We don't specify confs because each request can have a different conf above
+	// the minimum. So we do all conf handling in getConfirmedAt.
+	logs, err := lp.LogsWithSigs(
+		lastProcessedBlock,
+		latestBlock.BlockNumber,
```

Comment:
lookback stuff seems common, maybe we should consider adding LogsWithSigs query which only has a start block and goes to tip